src/tomography.jl

```diff
@@ -73,88 +73,68 @@ end
+    if status != :Optimal
+        println("FreeLSStateTomo fit return status: $status")
+    end
+    return ρest
 end
 
 """
-    QST_SDP(expResults, varMat, measPulseMap, measOpMap, measPulseUs, measOps, n)
+    QST_ML(expResults, varMat, measPulseMap, measOpMap, measPulseUs, measOps, n)
 
-Function to perform constrained SDP optimization of a physical density matrix
-consitent with the data.
+Function to perform maximum-likelihood quantum state tomography.
 
-   + expResults : structure array (length total number of experiments)
-                  each structure containts fields data, measPulse, measOperator
-   + measPulses : array of unitaries of measurement pulses
-   + measOps : array of measurement operators
-   + n : number of qubits
+   + expResults : data array
+   + varmat : covariance matrix for data
+   + measPulseMap : array mapping each experiment to a measurement readout
+     pulse
+   + measOpMap: array mapping each experiment to a measurement channel
+   + measPulseUs : array of unitaries of measurement pulses
+   + measOps : array of measurement operators for each channel
 """
-function QST_SDP(expResults, varMat, measPulseMap, measOpMap, measPulseUs, measOps, n)
-    #Construct the predictor matrix.  Each row is an experiment.  The number of
-    #columns is 4^n for the size of the vectorized density matrix.
-    #First transform the measurement operators by the readout pulses to create
-    #the effective measurement operators and then flatten into row of the
-    #predictor matrix
-
-    predictorMat = zeros(Complex128, length(expResults), 4^n)
-    for expct = 1:length(expResults)
-        tmp = transpose(measPulseUs[measPulseMap[expct]]'*measOps[measOpMap[expct]]*measPulseUs[measPulseMap[expct]])
-        predictorMat[expct,:] = tmp[:]
+function QST_ML(expResults, varMat, measPulseMap, measOpMap, measPulseUs, measOps)
+    # construct the vector of observables for each experiment
+    obs = Matrix{Complex128}[]
+    for ct in 1:length(expResults)
+        U = measPulseUs[measPulseMap[ct]]
+        op = measOps[measOpMap[ct]]
+        push!(obs, U' * op * U)
     end
+    tomo = MLStateTomo(obs)
 
-    # solver = SCSSolver(verbose=0, max_iters=10_000, eps = 1e-8)
-
-    # invVarMat = inv(varMat);
-    # invVarMat = real(sqrtm(real(sqrtm(invVarMat'*invVarMat))));
-
-    # ρr = Variable(2^n, 2^n)
-    # ρi = Variable(2^n, 2^n)
-
-    # constraints = trace(ρr) == 1
-    # constraints += trace(ρi) == 0
-    # constraints += isposdef([ρr ρi; -ρi ρr])
-
-    #We want to minimize the difference between predicted results and experimental results
-    # problem = minimize( vecnorm(invVarMat*(expResults - [real(predictorMat) imag(predictorMat)]*[vec(ρr); vec(ρi)]), 2)^2, constraints )
-    # solve!(problem, solver)
-
-    println("Done")
-    # return (ρr.value - 1im*ρi.value)
-    return eye(2*n)
+    # TODO MLStateTomo does not currently support variance data
```

Comment:
Ok, I'll change this to use `LSStateTomo`. It would also be nice to add "Wizard" (i.e. Smolin) state tomography to `QuantumTomograph.jl`. I guess we stopped using that, though, because it didn't account for variance information.